zerver/models.py

```diff
@@ -387,7 +387,7 @@ def flush_realm_emoji(sender, **kwargs):
 
 def filter_pattern_validator(value):
     # type: (Text) -> None
-    regex = re.compile(r'(?:[\w\-#]+)(\(\?P<\w+>.+\))')
+    regex = re.compile(r'(?:[\w\-#]*)(\(\?P<\w+>.+\))')
     error_msg = 'Invalid filter pattern, you must use the following format PREFIX-(?P<id>.+)'
```

Comment:
We should probably tweak this error message to be a bit clearer given the new range of what's supported.